File: app4web_2022/app/src/main/java/site/app4web/app4web/Action/JasonSessionAction.java
package site.app4web.app4web.Action;

import android.content.Context;
import android.content.SharedPreferences;
import android.util.Log;
import site.app4web.app4web.Helper.JasonHelper;
import org.json.JSONObject;
import java.net.URI;

public class JasonSessionAction {
    public void set(final JSONObject action, final JSONObject data, final JSONObject event, final Context context){
        try {
            JSONObject options = action.getJSONObject("options");
            String domain;
            if(options.has("domain")){
                String urlString = options.getString("domain");
                if(!urlString.startsWith("http")){
                    urlString = "https://" + urlString;
                }
                URI uri = new URI(urlString);
                domain = uri.getHost().toLowerCase();
            }else if(options.has("url")){
                String urlString = options.getString("url");
                if(!urlString.startsWith("http")){
                    urlString = "https://" + urlString;
                }
                URI uri = new URI(urlString);
                domain = uri.getHost().toLowerCase();
            } else {
                return;
            }

            // store either header or body under the domain name
            // сохраняем заголовок или тело под доменным именем
            SharedPreferences pref = context.getSharedPreferences("session", 0);
            SharedPreferences.Editor editor = pref.edit();

            // Stringify object first
            // Stringify объект первым
            String stringified_session = options.toString();
            editor.putString(domain, stringified_session);
            editor.commit();

            JasonHelper.next("success", action, new JSONObject(), event, context);

        } catch (Exception e){
            Log.d("Warning", e.getStackTrace()[0].getMethodName() + " : " + e.toString());
        }


    }
    public void reset(final JSONObject action, final JSONObject data, final JSONObject event, final Context context){
        try {
            JSONObject options = action.getJSONObject("options");
            String domain;
            if(options.has("domain")){
                String urlString = options.getString("domain");
                if(!urlString.startsWith("http")){
                    urlString = "https://" + urlString;
                }
                URI uri = new URI(urlString);
                domain = uri.getHost().toLowerCase();
            }else if(options.has("url")){
                String urlString = options.getString("url");
                if(!urlString.startsWith("http")){
                    urlString = "https://" + urlString;
                }
                URI uri = new URI(urlString);
                domain = uri.getHost().toLowerCase();
            } else {
                return;
            }

            SharedPreferences pref = context.getSharedPreferences("session", 0);
            SharedPreferences.Editor editor = pref.edit();
            editor.remove(domain);
            editor.commit();

            JasonHelper.next("success", action, new JSONObject(), event, context);

        } catch (Exception e){
            Log.d("Warning", e.getStackTrace()[0].getMethodName() + " : " + e.toString());
        }
    }
}

File: app4web_k2/app/src/main/java/site/app4web/app4web/Action/JasonSessionAction.kt
package site.app4web.app4web.Action

import android.content.Context
import android.util.Log
import org.json.JSONObject
import site.app4web.app4web.Helper.JasonHelper
import java.net.URI


class JasonSessionAction {
    operator fun set(action: JSONObject, data: JSONObject?, event: JSONObject?, context: Context) {
        try {
            val options = action.getJSONObject("options")
            val domain: String
            if (options.has("domain")) {
                var urlString = options.getString("domain")
                if (!urlString.startsWith("http")) {
                    urlString = "https://$urlString"
                }
                val uri = URI(urlString)
                domain = uri.host.toLowerCase()
            } else if (options.has("url")) {
                var urlString = options.getString("url")
                if (!urlString.startsWith("http")) {
                    urlString = "https://$urlString"
                }
                val uri = URI(urlString)
                domain = uri.host.toLowerCase()
            } else {
                return
            }

            // store either header or body under the domain name
            // сохраняем заголовок или тело под доменным именем
            val pref = context.getSharedPreferences("session", 0)
            val editor = pref.edit()

            // Stringify object first
            // Stringify объект первым
            val stringified_session = options.toString()
            editor.putString(domain, stringified_session)
            editor.commit()
            JasonHelper.next("success", action, JSONObject(), event, context)
        } catch (e: Exception) {
            Log.d("Warning", e.stackTrace[0].methodName + " : " + e.toString())
        }
    }

    fun reset(action: JSONObject, data: JSONObject?, event: JSONObject?, context: Context) {
        try {
            val options = action.getJSONObject("options")
            val domain: String
            if (options.has("domain")) {
                var urlString = options.getString("domain")
                if (!urlString.startsWith("http")) {
                    urlString = "https://$urlString"
                }
                val uri = URI(urlString)
                domain = uri.host.toLowerCase()
            } else if (options.has("url")) {
                var urlString = options.getString("url")
                if (!urlString.startsWith("http")) {
                    urlString = "https://$urlString"
                }
                val uri = URI(urlString)
                domain = uri.host.toLowerCase()
            } else {
                return
            }
            val pref = context.getSharedPreferences("session", 0)
            val editor = pref.edit()
            editor.remove(domain)
            editor.commit()
            JasonHelper.next("success", action, JSONObject(), event, context)
        } catch (e: Exception) {
            Log.d("Warning", e.stackTrace[0].methodName + " : " + e.toString())
        }
    }
}